### PR TITLE
Feat cart auth

### PR DESCRIPTION
**Pull request by JowelTisso**
This PR is for cart auth feature:
* Cart is moved from local state to user api
* Clear data feature on logout
* Minor changes in other files

### CODE REVIEW

File: src/components/productCard/ProductCard.jsx [hunk 1]
```diff
@@ -45,7 +49,18 @@ const ProductCard = ({
   };
 
   const addToCartHandler = () => {
-    addToCart(data, cartDispatch);
+    try {
+      const itemIndex = cartState?.cartItems?.findIndex(
```
Comment (Ash-D23) [inline src/components/productCard/ProductCard.jsx]: you can modularize this further by creating a function to check item in cart true or false

File: src/pages/products/helper/CartHelper.js [hunk 2]
```diff
@@ -1,32 +1,81 @@
...
+  try {
+    const token = getUserToken();
+    if (token) {
+      const { status, data } = await axios.post(
+        "/api/user/cart",
+        {
+          product: item,
+        },
+        {
+          headers: { authorization: token },
```
Comment (Ash-D23) [inline src/pages/products/helper/CartHelper.js:16]: you can put headers object globally so that in each function you dont have to setup headers

**Review by Ash-D23 — APPROVED** [review]
- Clean Code
-  Good use of Context and API Calls
-  Easy to read code

File: src/pages/products/helper/CartHelper.js (hunk 2 above)
Comment (JowelTisso) [reply src/pages/products/helper/CartHelper.js:16]: Ok, will try that

File: src/components/productCard/ProductCard.jsx (hunk 1 above)
Comment (JowelTisso) [reply src/components/productCard/ProductCard.jsx]: Ok, will try that

**Review by narharikale — COMMENTED** [review]
i am able to add same product in wishlist  many times

**Comment by JowelTisso** [comment]
> i am able to add same product in wishlist many times

Ok, I got it, when I click the add to wishlist icon fast enough then two items gets added

**Review by SukanyaSen530 — APPROVED** [review]

**Pull request merged**